lib/kubeclient.rb

```diff
@@ -22,8 +22,12 @@ class Client
       [Kubeclient.const_set(et, Class.new(RecursiveOpenStruct)), et]
     end
 
-    def initialize(uri, version)
+    def initialize(uri, version = 'v1beta3')
       @api_endpoint = (uri.is_a? URI) ? uri : URI.parse(uri)
+
+      unless @api_endpoint.path.include? 'api'
+        @api_endpoint.merge!(File.join(@api_endpoint.path, 'api'))
```

Comment:
Actually since you have a URI object,  perhaps you can just use it,  which
would be the most accurate.
On Mar 19, 2015 6:12 AM, "abonas" <notifications@github.com> wrote:

> In lib/kubeclient.rb
> <https://github.com/abonas/kubeclient/pull/42#discussion_r26741641>:
>
> >        @api_endpoint = (uri.is_a? URI) ? uri : URI.parse(uri)
> > +
> > +      unless @api_endpoint.path.include? 'api'
> > +        @api_endpoint.merge!(File.join(@api_endpoint.path, 'api'))
>
>  You probably don't want this because File.join will use the default file
> separator which is \ on Windows. I would just use string interpolation.
>
> good point, thanks !
> I assume the same goes for line 31 too (which is not part of this patch
> though)
>
> BTW from looking at the documentation of the method it's really unclear...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/abonas/kubeclient/pull/42/files#r26741641>.
>